x/operator/keeper/operator.go

```diff
@@ -0,0 +1,117 @@
+	}
+	// todo: to check the validation of input info
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), operatortypes.KeyPrefixOperatorInfo)
+	// todo: think about the difference between init and update in future
+
+	// key := common.HexToAddress(incentive.Contract)
+	bz := k.cdc.MustMarshal(info)
+
+	store.Set(opAccAddr, bz)
+	return nil
+}
+
+func (k *Keeper) OperatorInfo(ctx sdk.Context, addr string) (info *operatortypes.OperatorInfo, err error) {
+	opAccAddr, err := sdk.AccAddressFromBech32(addr)
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "GetOperatorInfo: error occurred when parse acc address from Bech32")
+	}
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), operatortypes.KeyPrefixOperatorInfo)
+	// key := common.HexToAddress(incentive.Contract)
+	isExist := store.Has(opAccAddr)
+	if !isExist {
+		return nil, errorsmod.Wrap(operatortypes.ErrNoKeyInTheStore, fmt.Sprintf("GetOperatorInfo: key is %suite", opAccAddr))
+	}
+
+	value := store.Get(opAccAddr)
+
+	ret := operatortypes.OperatorInfo{}
+	k.cdc.MustUnmarshal(value, &ret)
+	return &ret, nil
+}
+
+func (k *Keeper) IsOperator(ctx sdk.Context, addr sdk.AccAddress) bool {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), operatortypes.KeyPrefixOperatorInfo)
+	return store.Has(addr)
+}
+
+func (k *Keeper) UpdateOptedInfo(ctx sdk.Context, operatorAddr, avsAddr string, info *operatortypes.OptedInfo) error {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), operatortypes.KeyPrefixOperatorOptedAVSInfo)
+
+	// check operator address validation
+	_, err := sdk.AccAddressFromBech32(operatorAddr)
+	if err != nil {
+		return restakingtype.ErrOperatorAddr
+	}
+	infoKey := restakingtype.GetJoinedStoreKey(operatorAddr, avsAddr)
+
+	bz := k.cdc.MustMarshal(info)
+	store.Set(infoKey, bz)
+	return nil
+}
+
+func (k *Keeper) GetOptedInfo(ctx sdk.Context, operatorAddr, avsAddr string) (info *operatortypes.OptedInfo, err error) {
+	opAccAddr, err := sdk.AccAddressFromBech32(operatorAddr)
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "GetOptedInfo: error occurred when parse acc address from Bech32")
+	}
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), operatortypes.KeyPrefixOperatorOptedAVSInfo)
+	infoKey := restakingtype.GetJoinedStoreKey(operatorAddr, avsAddr)
+	ifExist := store.Has(infoKey)
+	if !ifExist {
+		return nil, errorsmod.Wrap(operatortypes.ErrNoKeyInTheStore, fmt.Sprintf("GetOptedInfo: key is %suite", opAccAddr))
+	}
+
+	value := store.Get(infoKey)
+
+	ret := operatortypes.OptedInfo{}
+	k.cdc.MustUnmarshal(value, &ret)
+	return &ret, nil
+}
+
+func (k *Keeper) IsOptedIn(ctx sdk.Context, operatorAddr, avsAddr string) bool {
+	optedInfo, err := k.GetOptedInfo(ctx, operatorAddr, avsAddr)
+	if err != nil {
+		return false
+	}
+	if optedInfo.OptedOutHeight != operatortypes.DefaultOptedOutHeight {
+		return false
+	}
+	return true
+}
+
+func (k *Keeper) GetOptedInAVSForOperator(ctx sdk.Context, operatorAddr string) ([]string, error) {
```

Comment:
This function should also return the app chains / dogfood chain?

Reply:
Yeah, It returns a slice of `avsAddr`. As mentioned above, the `avsAddr` should be a universal identifier for all AVS, including app chain AVS.